src/Network/Broadcast/OutboundQueue.hs

```diff
@@ -295,21 +295,18 @@ defaultEnqueuePolicyEdgeP2P = go
   where
     -- Enqueue policy for edge nodes
     go :: EnqueuePolicy nid
-    go (MsgTransaction OriginSender) = [
+    go (MsgTransaction _) = [
         EnqueueAll NodeRelay (MaxAhead 3) PLow
       ]
-    go (MsgTransaction (OriginForward _)) = [
-        -- don't forward transactions that weren't created at this node
-      ]
     go (MsgAnnounceBlockHeader _) = [
         EnqueueAll NodeRelay (MaxAhead 0) PHighest
       ]
     go MsgRequestBlockHeaders = [
         EnqueueAll NodeRelay (MaxAhead 1) PHigh
       ]
-    go (MsgRequestBlock _) = [
+    go (MsgRequestBlocks _) = [
         -- Edge nodes can only talk to relay nodes
-        EnqueueOne [NodeRelay] UnlimitedMaxAhead PHigh
+        EnqueueOne [NodeRelay] (MaxAhead 1) PHigh
```

Comment:
Because p2p nodes *do* send block header announcements, which are higher priority than requests for blocks.